experimental/oak_baremetal_kernel/src/args.rs

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use atomic_refcell::AtomicRefCell;
+use core::ffi::CStr;
+use lazy_static::lazy_static;
+
+// This is a really crude analogue of CString, which we can't use, as CString requires allocation.
+// Instead of allocation, we rely on a static buffer.
+struct StaticCString {
```

Comment:
I wonder if https://docs.rs/arrayvec/0.7.2/arrayvec/struct.ArrayString.html may help here?

Reply:
Yep, that's much better!